src/types.ts

```diff
@@ -0,0 +1,6 @@
+import { LifetimeType } from './lifetime'
+
+export type ResolutionStack = Array<{
```

Comment:
We could probably put this type definition inside of `container.ts`, it feels a bit alone here. Alternatively, we could refactor the type definitions inside of `container.ts` to here, assuming we export them in `index.ts`. What do you think?

Reply:
That makes sense. I only did this as a quick hack to remove `ResolutionStack` from the public API surface while avoiding having to manually specify all exports from `./container.ts` (since the index has `export *`).

The proper solution is to go through and rather than using wildcard export, explicitly export only those items which are intended to be in the public API.

Reply:
Would this type need to be exported if put in `container.ts` though?

Reply:
Nope, I just went through and cleaned up all the wildcard exports to match what (I think) the public API should be.

Reply:
Took a peek and I noticed `RegistrationHash` is a public interface but isn't exported. Could there be others?